sherpa/astro/data.py

```diff
@@ -123,13 +123,14 @@
 from collections.abc import Sequence
 import logging
 import os
-from typing import Any, Callable, Literal, Mapping, cast, overload
+from typing import Any, Callable, Literal, Mapping, cast, overload, TYPE_CHECKING
```

Comment:
This is not a full review.

We should change line 123 to 

```python
from collections.abc import Callable, Mapping, Sequence
```

rather than import the `Callable` and `Mapping` symbols from `typing`

I'm 100% sure about `Callable` and slightly less sure about `Mapping` ....

In some of the code below you accept a `dict | None` and perhaps that should be `Mapping | None` or `Mapping[str, Any] | None` ?